pkg/event/registry.go

```diff
@@ -0,0 +1,142 @@
+package event
+
+import (
+	"github.com/pkg/errors"
+
+	k8sV1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog"
+
+	"github.com/submariner-io/admiral/pkg/log"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+)
+
+type Registry struct {
+	name          string
+	networkPlugin string
+	eventHandlers []Handler
+}
+
+// NewRegistry creates a new registry, the name would normally be the name of the owner, for example
+// 'routeagent', and the networkPlugin must be the network plugin that has been detected (or passed
+// via env-var) this will be used to only keep the relevant handlers in runtime, while discarding
+// the ones that shall not be ran with the system network plugin.
+func NewRegistry(name, networkPlugin string) Registry {
+	return Registry{
+		name:          name,
+		networkPlugin: networkPlugin,
+		eventHandlers: []Handler{},
+	}
+}
+
+func (er *Registry) addHandler(eventHandler Handler) {
+	evNetworkPlugin := eventHandler.GetNetworkPlugin()
+
+	if evNetworkPlugin == AnyNetworkPlugin || evNetworkPlugin == er.networkPlugin {
```

Comment:
Instead of using an empty value for "AnyNetworkPlugin", shall we use "All" (or "Any")?

Reply:
It was a previous suggestion by @tpantelis, either way works for me.

Reply:
Typically empty is used for such things to eliminate the chance for a clash with a real value, however unlikely that is.